test-lambda.sh uses optimised by default but shadowJar does not generate optimized JAR

```
./test-lambda.sh              

> Task :app:compileJava
Note: Creating bean classes for 1 type elements

Deprecated Gradle features were used in this build, making it incompatible with Gradle 9.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/8.2.1/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 2s
3 actionable tasks: 3 executed

> Task :app:compileTestJava
Note: Creating bean classes for 1 type elements

Deprecated Gradle features were used in this build, making it incompatible with Gradle 9.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/8.2.1/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 1s
8 actionable tasks: 6 executed, 2 up-to-date

> Task :infra:run
Exception in thread "main" software.amazon.jsii.JsiiError: Cannot find asset at /Users/sdelamo/github/softamo/projectcheckins/app/build/libs/app-0.1-all-optimized.jar
Error: Cannot find asset at /Users/sdelamo/github/softamo/projectcheckins/app/build/libs/app-0.1-all-optimized.jar
    at new AssetStaging (/var/folders/65/67w3r16d2cn0jmpygf_mgch80000gn/T/jsii-kernel-uVcS4Q/node_modules/aws-cdk-lib/core/lib/asset-staging.js:1:1402)
    at new Asset (/var/folders/65/67w3r16d2cn0jmpygf_mgch80000gn/T/jsii-kernel-uVcS4Q/node_modules/aws-cdk-lib/aws-s3-assets/lib/asset.js:1:736)
    at AssetCode.bind (/var/folders/65/67w3r16d2cn0jmpygf_mgch80000gn/T/jsii-kernel-uVcS4Q/node_modules/aws-cdk-lib/aws-lambda/lib/code.js:1:4628)
    at new Function (/var/folders/65/67w3r16d2cn0jmpygf_mgch80000gn/T/jsii-kernel-uVcS4Q/node_modules/aws-cdk-lib/aws-lambda/lib/function.js:1:7479)
    at Kernel._Kernel_create (/private/var/folders/65/67w3r16d2cn0jmpygf_mgch80000gn/T/jsii-java-runtime15254390170405739726/lib/program.js:10071:25)
    at Kernel.create (/private/var/folders/65/67w3r16d2cn0jmpygf_mgch80000gn/T/jsii-java-runtime15254390170405739726/lib/program.js:9742:93)
    at KernelHost.processRequest (/private/var/folders/65/67w3r16d2cn0jmpygf_mgch80000gn/T/jsii-java-runtime15254390170405739726/lib/program.js:11658:36)
    at KernelHost.run (/private/var/folders/65/67w3r16d2cn0jmpygf_mgch80000gn/T/jsii-java-runtime15254390170405739726/lib/program.js:11618:22)
    at Immediate._onImmediate (/private/var/folders/65/67w3r16d2cn0jmpygf_mgch80000gn/T/jsii-java-runtime15254390170405739726/lib/program.js:11619:46)
    at process.processImmediate (node:internal/timers:476:21)
        at software.amazon.jsii.JsiiRuntime.processErrorResponse(JsiiRuntime.java:150)
        at software.amazon.jsii.JsiiRuntime.requestResponse(JsiiRuntime.java:116)
        at software.amazon.jsii.JsiiClient.createObject(JsiiClient.java:89)
        at software.amazon.jsii.JsiiEngine.createNewObject(JsiiEngine.java:614)
        at software.amazon.awscdk.services.lambda.Function.<init>(Function.java:53)
        at software.amazon.awscdk.services.lambda.Function$Builder.build(Function.java:1199)
        at com.softamo.projectcheckins.AppStack.<init>(AppStack.java:51)
        at com.softamo.projectcheckins.AppStack.<init>(AppStack.java:29)
        at com.softamo.projectcheckins.Main.main(Main.java:8)


> Task :infra:run FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':infra:run'.
> Process 'command '/Users/sdelamo/.sdkman/candidates/java/17.0.8-amzn/bin/java'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 9.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/8.2.1/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 2s
2 actionable tasks: 1 executed, 1 up-to-date

Subprocess exited with error 1

> Task :infra:run FAILED
Exception in thread "main" software.amazon.jsii.JsiiError: Cannot find asset at /Users/sdelamo/github/softamo/projectcheckins/app/build/libs/app-0.1-all-optimized.jar
Error: Cannot find asset at /Users/sdelamo/github/softamo/projectcheckins/app/build/libs/app-0.1-all-optimized.jar
    at new AssetStaging (/var/folders/65/67w3r16d2cn0jmpygf_mgch80000gn/T/jsii-kernel-XfMKEp/node_modules/aws-cdk-lib/core/lib/asset-staging.js:1:1402)
    at new Asset (/var/folders/65/67w3r16d2cn0jmpygf_mgch80000gn/T/jsii-kernel-XfMKEp/node_modules/aws-cdk-lib/aws-s3-assets/lib/asset.js:1:736)
    at AssetCode.bind (/var/folders/65/67w3r16d2cn0jmpygf_mgch80000gn/T/jsii-kernel-XfMKEp/node_modules/aws-cdk-lib/aws-lambda/lib/code.js:1:4628)
    at new Function (/var/folders/65/67w3r16d2cn0jmpygf_mgch80000gn/T/jsii-kernel-XfMKEp/node_modules/aws-cdk-lib/aws-lambda/lib/function.js:1:7479)
    at Kernel._Kernel_create (/private/var/folders/65/67w3r16d2cn0jmpygf_mgch80000gn/T/jsii-java-runtime16554925817108921386/lib/program.js:10071:25)
    at Kernel.create (/private/var/folders/65/67w3r16d2cn0jmpygf_mgch80000gn/T/jsii-java-runtime16554925817108921386/lib/program.js:9742:93)
    at KernelHost.processRequest (/private/var/folders/65/67w3r16d2cn0jmpygf_mgch80000gn/T/jsii-java-runtime16554925817108921386/lib/program.js:11658:36)
    at KernelHost.run (/private/var/folders/65/67w3r16d2cn0jmpygf_mgch80000gn/T/jsii-java-runtime16554925817108921386/lib/program.js:11618:22)
    at Immediate._onImmediate (/private/var/folders/65/67w3r16d2cn0jmpygf_mgch80000gn/T/jsii-java-runtime16554925817108921386/lib/program.js:11619:46)
    at process.processImmediate (node:internal/timers:476:21)
        at software.amazon.jsii.JsiiRuntime.processErrorResponse(JsiiRuntime.java:150)
        at software.amazon.jsii.JsiiRuntime.requestResponse(JsiiRuntime.java:116)
        at software.amazon.jsii.JsiiClient.createObject(JsiiClient.java:89)
        at software.amazon.jsii.JsiiEngine.createNewObject(JsiiEngine.java:614)
        at software.amazon.awscdk.services.lambda.Function.<init>(Function.java:53)
        at software.amazon.awscdk.services.lambda.Function$Builder.build(Function.java:1199)
        at com.softamo.projectcheckins.AppStack.<init>(AppStack.java:51)
        at com.softamo.projectcheckins.AppStack.<init>(AppStack.java:29)
        at com.softamo.projectcheckins.Main.main(Main.java:8)
```

--- a/starter-aws-cdk/src/main/java/io/micronaut/aws/cdk/function/MicronautFunction.java
+++ b/starter-aws-cdk/src/main/java/io/micronaut/aws/cdk/function/MicronautFunction.java
@@ -57,7 +57,7 @@ public final class MicronautFunction {
     }
 
     private static Runtime runtime(boolean graalVMNative) {
-        return graalVMNative ? Runtime.PROVIDED_AL2 : Runtime.JAVA_11;
+        return graalVMNative ? Runtime.PROVIDED_AL2 : Runtime.JAVA_17;
     }
 
 }

--- a/starter-aws-cdk/src/test/groovy/io/micronaut/aws/cdk/function/MicronautFunctionSpec.groovy
+++ b/starter-aws-cdk/src/test/groovy/io/micronaut/aws/cdk/function/MicronautFunctionSpec.groovy
@@ -44,8 +44,8 @@ class MicronautFunctionSpec extends Specification {
         where:
         applicationType          | graalVMNative | runtime                | handler
         ApplicationType.DEFAULT  | true          | Runtime.PROVIDED_AL2   | null
-        ApplicationType.DEFAULT  | false         | Runtime.JAVA_11        | null
+        ApplicationType.DEFAULT  | false         | Runtime.JAVA_17        | null
         ApplicationType.FUNCTION | true          | Runtime.PROVIDED_AL2   | "example.micronaut.MyHandler"
-        ApplicationType.FUNCTION | false         | Runtime.JAVA_11        | "example.micronaut.MyHandler"
+        ApplicationType.FUNCTION | false         | Runtime.JAVA_17        | "example.micronaut.MyHandler"
     }
 }

--- a/starter-core/src/main/java/io/micronaut/starter/feature/aws/Cdk.java
+++ b/starter-core/src/main/java/io/micronaut/starter/feature/aws/Cdk.java
@@ -130,11 +130,19 @@ public class Cdk implements MultiProjectFeature, InfrastructureAsCodeFeature {
     public String getCategory() {
         return Category.CLOUD;
     }
-
+    
     @Override
     public void apply(GeneratorContext generatorContext) {
+        boolean optimized = generatorContext.getFeatures().hasFeature(MicronautAot.class);
+        boolean nativeImage = generatorContext.getFeatures().hasFeature(GraalVM.class);
         if (generatorContext.getFeatures().hasFeature(AwsApiFeature.class)) {
-            generatorContext.addTemplate("test-lambda", new RockerTemplate(Template.ROOT, "test-lambda.sh", testlambda.template(generatorContext.getBuildTool(), generatorContext.getFeatures().hasFeature(GraalVM.class), INFRA_MODULE), true));
+            generatorContext.addTemplate("test-lambda", new RockerTemplate(Template.ROOT,
+                    "test-lambda.sh",
+                    testlambda.template(generatorContext.getBuildTool(),
+                            nativeImage,
+                            optimized,
+                            INFRA_MODULE),
+                    true));
         }
         generatorContext.addTemplate("cdk-json", new RockerTemplate(INFRA_MODULE, "cdk.json", cdkjson.template(generatorContext.getBuildTool(), INFRA_MODULE)));
         generatorContext.addTemplate("cdk-main", new RockerTemplate(INFRA_MODULE, "src/main/java/{packagePath}/" + MAIN_CLASS_NAME + ".java",
@@ -157,8 +165,8 @@ public class Cdk implements MultiProjectFeature, InfrastructureAsCodeFeature {
                         generatorContext.getFeatures().hasFeature(AwsApiFeature.class) ? "micronaut-function-api" : null,
                         "0.1",
                         handler,
-                        generatorContext.getFeatures().hasFeature(GraalVM.class),
-                        generatorContext.getFeatures().hasFeature(MicronautAot.class),
+                        nativeImage,
+                        optimized,
                         generatorContext.getJdkVersion()))
         );
         buildRockerModel(generatorContext).ifPresent(rockerModel -> {
@@ -166,7 +174,7 @@ public class Cdk implements MultiProjectFeature, InfrastructureAsCodeFeature {
                     new RockerTemplate(INFRA_MODULE, generatorContext.getBuildTool().getBuildFileName(), rockerModel));
         });
 
-        generatorContext.addHelpTemplate(new RockerWritable(cdkhelp.template(generatorContext.getBuildTool(), generatorContext.getFeatures().hasFeature(GraalVM.class), INFRA_MODULE)));
+        generatorContext.addHelpTemplate(new RockerWritable(cdkhelp.template(generatorContext.getBuildTool(), generatorContext.getFeatures().hasFeature(GraalVM.class), generatorContext.getFeatures().hasFeature(MicronautAot.class), INFRA_MODULE)));
     }
 
     protected void addAppStackTest(@NonNull GeneratorContext generatorContext,

--- a/starter-core/src/main/java/io/micronaut/starter/feature/aws/template/cdkappstack.rocker.raw
+++ b/starter-core/src/main/java/io/micronaut/starter/feature/aws/template/cdkappstack.rocker.raw
@@ -258,7 +258,7 @@ public class AppStack extends Stack {
 
     public static String functionFilename() {
         return MicronautFunctionFile.builder()
-@if (optimized) {
+@if (!nativeImage && optimized) {
             .optimized()
 }
             .graalVMNative(@nativeImage)

--- a/starter-core/src/main/java/io/micronaut/starter/feature/aws/template/cdkhelp.rocker.raw
+++ b/starter-core/src/main/java/io/micronaut/starter/feature/aws/template/cdkhelp.rocker.raw
@@ -1,5 +1,5 @@
 @import io.micronaut.starter.options.BuildTool;
-@args(BuildTool buildTool, boolean nativeImage, String infrastructureModuleName)
+@args(BuildTool buildTool, boolean nativeImage, boolean optimized, String infrastructureModuleName)
 
 ## Requisites
 
@@ -14,6 +14,9 @@
 ```
 @if(buildTool.isGradle() && nativeImage) {
 ./gradlew :app:buildNativeLambda
+./gradlew test
+} else if(buildTool.isGradle() && !nativeImage && optimized) {
+./gradlew :app:optimizedJitJarAll
 ./gradlew test
 } else if(buildTool.isGradle() && !nativeImage) {
 ./gradlew :app:shadowJar

--- a/starter-core/src/main/java/io/micronaut/starter/feature/aws/template/testlambda.rocker.raw
+++ b/starter-core/src/main/java/io/micronaut/starter/feature/aws/template/testlambda.rocker.raw
@@ -17,6 +17,10 @@ fi
 if [ $EXIT_STATUS -ne 0 ]; then
   exit $EXIT_STATUS
 fi
+./gradlew test || EXIT_STATUS=$?
+if [ $EXIT_STATUS -ne 0 ]; then
+  exit $EXIT_STATUS
+fi
 } else if(buildTool.isGradle() && !nativeImage) {
 ./gradlew :app:shadowJar || EXIT_STATUS=$?
 if [ $EXIT_STATUS -ne 0 ]; then

--- a/starter-core/src/main/java/io/micronaut/starter/feature/aws/template/testlambda.rocker.raw
+++ b/starter-core/src/main/java/io/micronaut/starter/feature/aws/template/testlambda.rocker.raw
@@ -1,5 +1,5 @@
 @import io.micronaut.starter.options.BuildTool;
-@args(BuildTool buildTool, boolean nativeImage, String infrastructureModuleName)
+@args(BuildTool buildTool, boolean nativeImage, boolean optimized, String infrastructureModuleName)
 
 #!/bin/bash
 EXIT_STATUS=0
@@ -9,6 +9,11 @@ if [ $EXIT_STATUS -ne 0 ]; then
   exit $EXIT_STATUS
 fi
 ./gradlew test || EXIT_STATUS=$?
+if [ $EXIT_STATUS -ne 0 ]; then
+  exit $EXIT_STATUS
+fi
+} else if(buildTool.isGradle() && !nativeImage && optimized) {
+./gradlew :app:optimizedJitJarAll || EXIT_STATUS=$?
 if [ $EXIT_STATUS -ne 0 ]; then
   exit $EXIT_STATUS
 fi

--- a/test-lambda-graalvm.sh
+++ b/test-lambda-graalvm.sh
@@ -54,56 +54,6 @@ cd ../../..
 rm -rf starter-cli/temp
 
 ############
-# | BUILD  | TYPE     | RUNTIME | FEATURES
-# | MAVEN | FUNCTION  | PROVIDED    | aws-lambda,aws-cdk,aws-lambda-function-url,graalvm
-############
-
-# ./gradlew micronaut-cli:run --args="create-function-app -b maven -f \"aws-lambda,aws-cdk,aws-lambda-function-url,graalvm\" temp" || EXIT_STATUS=$?
-# 
-# if [ $EXIT_STATUS -ne 0 ]; then
-#   exit $EXIT_STATUS
-# fi
-# 
-# cd starter-cli/temp
-# 
-# ./test-lambda.sh  || EXIT_STATUS=$? 
-# 
-# if [ $EXIT_STATUS -ne 0 ]; then
-#   echo "❌ FAILED | MAVEN | FUNCTION  | PROVIDED    | aws-lambda,aws-cdk,aws-lambda-function-url,graalvm"
-#   exit $EXIT_STATUS
-# fi
-# 
-# cd infra
-# cdk destroy -f
-# cd ../../..
-# rm -rf starter-cli/temp
-
-############
-# | BUILD  | TYPE | RUNTIME  | FEATURES
-# | MAVEN  | APP  | PROVIDED | aws-lambda,aws-cdk,graalvm,amazon-api-gateway
-############
-
-# ./gradlew micronaut-cli:run --args="create-app -b maven -f \"aws-lambda,aws-cdk,graalvm,amazon-api-gateway\" temp" || EXIT_STATUS=$?
-# 
-# if [ $EXIT_STATUS -ne 0 ]; then
-#   exit $EXIT_STATUS
-# fi
-# 
-# cd starter-cli/temp
-# 
-# ./test-lambda.sh  || EXIT_STATUS=$? 
-# 
-# if [ $EXIT_STATUS -ne 0 ]; then
-#   echo "❌ FAILED | MAVEN  | APP  | PROVIDED | aws-lambda,aws-cdk,graalvm,amazon-api-gateway"
-#   exit $EXIT_STATUS
-# fi
-# 
-# cd infra
-# cdk destroy -f
-# cd ../../..
-# rm -rf starter-cli/temp
-
-############
 # | BUILD  | TYPE | RUNTIME  | FEATURES
 # | GRADLE | APP  | PROVIDED | aws-lambda,aws-cdk,graalvm,amazon-api-gateway
 ############
@@ -127,7 +77,6 @@ cd infra
 cdk destroy -f
 cd ../../..
 rm -rf starter-cli/temp
-
 
 ############
 # | BUILD  | TYPE | RUNTIME  | FEATURES
@@ -204,30 +153,5 @@ cd infra
 cdk destroy -f
 cd ../../..
 rm -rf starter-cli/temp
-
-############
-# | BUILD  | TYPE     | RUNTIME | FEATURES
-# | MAVEN | FUNCTION  | YES    | aws-lambda,aws-cdk,amazon-api-gateway
-############
-
-# ./gradlew micronaut-cli:run --args="create-function-app -b maven -f \"aws-lambda,aws-cdk,graalvm,amazon-api-gateway\" temp" || EXIT_STATUS=$?
-# 
-# if [ $EXIT_STATUS -ne 0 ]; then
-#   exit $EXIT_STATUS
-# fi
-# 
-# cd starter-cli/temp
-# 
-# ./test-lambda.sh  || EXIT_STATUS=$? 
-# 
-# if [ $EXIT_STATUS -ne 0 ]; then
-#   echo "❌ FAILED | MAVEN  | FUNCTION  | PROVIDED | aws-lambda,aws-cdk,graalvm,amazon-api-gateway"
-#   exit $EXIT_STATUS
-# fi
-# 
-# cd infra
-# cdk destroy -f
-# cd ../../..
-# rm -rf starter-cli/temp
 
 exit $EXIT_STATUS

--- a/test-lambda-maven-graalvm.sh
+++ b/test-lambda-maven-graalvm.sh
@@ -1,0 +1,82 @@
+#!/bin/bash
+EXIT_STATUS=0
+
+rm -rf starter-cli/temp
+
+############
+# | BUILD  | TYPE     | RUNTIME | FEATURES
+# | MAVEN | FUNCTION  | PROVIDED    | aws-lambda,aws-cdk,aws-lambda-function-url,graalvm
+############
+
+# ./gradlew micronaut-cli:run --args="create-function-app -b maven -f \"aws-lambda,aws-cdk,aws-lambda-function-url,graalvm\" temp" || EXIT_STATUS=$?
+#
+# if [ $EXIT_STATUS -ne 0 ]; then
+#   exit $EXIT_STATUS
+# fi
+#
+# cd starter-cli/temp
+#
+# ./test-lambda.sh  || EXIT_STATUS=$?
+#
+# if [ $EXIT_STATUS -ne 0 ]; then
+#   echo "❌ FAILED | MAVEN | FUNCTION  | PROVIDED    | aws-lambda,aws-cdk,aws-lambda-function-url,graalvm"
+#   exit $EXIT_STATUS
+# fi
+#
+# cd infra
+# cdk destroy -f
+# cd ../../..
+# rm -rf starter-cli/temp
+
+############
+# | BUILD  | TYPE | RUNTIME  | FEATURES
+# | MAVEN  | APP  | PROVIDED | aws-lambda,aws-cdk,graalvm,amazon-api-gateway
+############
+
+# ./gradlew micronaut-cli:run --args="create-app -b maven -f \"aws-lambda,aws-cdk,graalvm,amazon-api-gateway\" temp" || EXIT_STATUS=$?
+#
+# if [ $EXIT_STATUS -ne 0 ]; then
+#   exit $EXIT_STATUS
+# fi
+#
+# cd starter-cli/temp
+#
+# ./test-lambda.sh  || EXIT_STATUS=$?
+#
+# if [ $EXIT_STATUS -ne 0 ]; then
+#   echo "❌ FAILED | MAVEN  | APP  | PROVIDED | aws-lambda,aws-cdk,graalvm,amazon-api-gateway"
+#   exit $EXIT_STATUS
+# fi
+#
+# cd infra
+# cdk destroy -f
+# cd ../../..
+# rm -rf starter-cli/temp
+
+
+############
+# | BUILD  | TYPE     | RUNTIME | FEATURES
+# | MAVEN | FUNCTION  | YES    | aws-lambda,aws-cdk,amazon-api-gateway
+############
+
+# ./gradlew micronaut-cli:run --args="create-function-app -b maven -f \"aws-lambda,aws-cdk,graalvm,amazon-api-gateway\" temp" || EXIT_STATUS=$?
+#
+# if [ $EXIT_STATUS -ne 0 ]; then
+#   exit $EXIT_STATUS
+# fi
+#
+# cd starter-cli/temp
+#
+# ./test-lambda.sh  || EXIT_STATUS=$?
+#
+# if [ $EXIT_STATUS -ne 0 ]; then
+#   echo "❌ FAILED | MAVEN  | FUNCTION  | PROVIDED | aws-lambda,aws-cdk,graalvm,amazon-api-gateway"
+#   exit $EXIT_STATUS
+# fi
+#
+# cd infra
+# cdk destroy -f
+# cd ../../..
+# rm -rf starter-cli/temp
+
+exit $EXIT_STATUS

--- a/test-lambda-maven.sh
+++ b/test-lambda-maven.sh
@@ -1,0 +1,107 @@
+#!/bin/bash
+EXIT_STATUS=0
+
+rm -rf starter-cli/temp
+
+############
+# | BUILD  | TYPE | RUNTIME | FEATURES
+# | MAVEN  | APP  | JAVA    | aws-lambda,aws-cdk,amazon-api-gateway-http
+############
+
+./gradlew micronaut-cli:run --args="create-app --jdk 17 -b maven -f \"aws-lambda,aws-cdk,amazon-api-gateway-http\" temp" || EXIT_STATUS=$?
+
+if [ $EXIT_STATUS -ne 0 ]; then
+  exit $EXIT_STATUS
+fi
+
+cd starter-cli/temp
+
+./test-lambda.sh  || EXIT_STATUS=$?
+
+if [ $EXIT_STATUS -ne 0 ]; then
+  echo "❌ FAILED | MAVEN  | APP  | JAVA | aws-lambda,aws-cdk,amazon-api-gateway-http"
+  exit $EXIT_STATUS
+fi
+
+cd infra
+cdk destroy -f
+cd ../../..
+rm -rf starter-cli/temp
+
+
+############
+# | BUILD  | TYPE     | RUNTIME | FEATURES
+# | MAVEN | FUNCTION  | JAVA    | aws-lambda,aws-cdk,aws-lambda-function-url
+############
+
+./gradlew micronaut-cli:run --args="create-function-app --jdk 17 -b maven -f \"aws-lambda,aws-cdk,aws-lambda-function-url\" temp" || EXIT_STATUS=$?
+
+if [ $EXIT_STATUS -ne 0 ]; then
+  exit $EXIT_STATUS
+fi
+
+cd starter-cli/temp
+
+./test-lambda.sh  || EXIT_STATUS=$?
+
+if [ $EXIT_STATUS -ne 0 ]; then
+  echo "❌ FAILED | MAVEN | FUNCTION  | JAVA    | aws-lambda,aws-cdk,aws-lambda-function-url"
+  exit $EXIT_STATUS
+fi
+
+cd infra
+cdk destroy -f
+cd ../../..
+rm -rf starter-cli/temp
+
+############
+# | BUILD  | TYPE | RUNTIME | FEATURES
+# | MAVEN  | APP  | JAVA    | aws-lambda,aws-cdk,amazon-api-gateway
+############
+
+./gradlew micronaut-cli:run --args="create-app --jdk 17 -b maven -f \"aws-lambda,aws-cdk,amazon-api-gateway\" temp" || EXIT_STATUS=$?
+
+if [ $EXIT_STATUS -ne 0 ]; then
+  exit $EXIT_STATUS
+fi
+
+cd starter-cli/temp
+
+./test-lambda.sh  || EXIT_STATUS=$?
+
+if [ $EXIT_STATUS -ne 0 ]; then
+  echo "❌ FAILED | MAVEN  | APP  | JAVA    | aws-lambda,aws-cdk,amazon-api-gateway"
+  exit $EXIT_STATUS
+fi
+
+cd infra
+cdk destroy -f
+cd ../../..
+rm -rf starter-cli/temp
+
+############
+# | BUILD  | TYPE     | RUNTIME | FEATURES
+# | MAVEN | FUNCTION  | JAVA    | aws-lambda,aws-cdk,amazon-api-gateway
+############
+
+./gradlew micronaut-cli:run --args="create-function-app --jdk 17 -b maven -f \"aws-lambda,aws-cdk,amazon-api-gateway\" temp" || EXIT_STATUS=$?
+
+if [ $EXIT_STATUS -ne 0 ]; then
+  exit $EXIT_STATUS
+fi
+
+cd starter-cli/temp
+
+./test-lambda.sh  || EXIT_STATUS=$?
+
+if [ $EXIT_STATUS -ne 0 ]; then
+  echo "❌ FAILED | MAVEN | FUNCTION  | JAVA    | aws-lambda,aws-cdk,amazon-api-gateway"
+  exit $EXIT_STATUS
+fi
+
+cd infra
+cdk destroy -f
+cd ../../..
+rm -rf starter-cli/temp
+
+exit $EXIT_STATUS

--- a/test-lambda.sh
+++ b/test-lambda.sh
@@ -35,31 +35,6 @@ cd ../../..
 rm -rf starter-cli/temp
 
 ############
-# | BUILD  | TYPE | RUNTIME | FEATURES
-# | MAVEN  | APP  | JAVA    | aws-lambda,aws-cdk,amazon-api-gateway-http
-############
-
-./gradlew micronaut-cli:run --args="create-app --jdk 17 -b maven -f \"aws-lambda,aws-cdk,amazon-api-gateway-http\" temp" || EXIT_STATUS=$?
-
-if [ $EXIT_STATUS -ne 0 ]; then
-  exit $EXIT_STATUS
-fi
-
-cd starter-cli/temp
-
-./test-lambda.sh  || EXIT_STATUS=$?
-
-if [ $EXIT_STATUS -ne 0 ]; then
-  echo "❌ FAILED | MAVEN  | APP  | JAVA | aws-lambda,aws-cdk,amazon-api-gateway-http"
-  exit $EXIT_STATUS
-fi
-
-cd infra
-cdk destroy -f
-cd ../../..
-rm -rf starter-cli/temp
-
-############
 # | BUILD  | TYPE     | RUNTIME | FEATURES
 # | GRADLE | FUNCTION | JAVA    | aws-lambda,aws-cdk,aws-lambda-function-url
 ############
@@ -76,57 +51,6 @@ cd starter-cli/temp
 
 if [ $EXIT_STATUS -ne 0 ]; then
   echo "❌ FAILED | GRADLE | FUNCTION | JAVA    | aws-lambda,aws-cdk,aws-lambda-function-url"
-  exit $EXIT_STATUS
-fi
-
-cd infra
-cdk destroy -f
-cd ../../..
-rm -rf starter-cli/temp
-
-
-############
-# | BUILD  | TYPE     | RUNTIME | FEATURES
-# | MAVEN | FUNCTION  | JAVA    | aws-lambda,aws-cdk,aws-lambda-function-url
-############
-
-./gradlew micronaut-cli:run --args="create-function-app --jdk 17 -b maven -f \"aws-lambda,aws-cdk,aws-lambda-function-url\" temp" || EXIT_STATUS=$?
-
-if [ $EXIT_STATUS -ne 0 ]; then
-  exit $EXIT_STATUS
-fi
-
-cd starter-cli/temp
-
-./test-lambda.sh  || EXIT_STATUS=$? 
-
-if [ $EXIT_STATUS -ne 0 ]; then
-  echo "❌ FAILED | MAVEN | FUNCTION  | JAVA    | aws-lambda,aws-cdk,aws-lambda-function-url"
-  exit $EXIT_STATUS
-fi
-
-cd infra
-cdk destroy -f
-cd ../../..
-rm -rf starter-cli/temp
-
-############
-# | BUILD  | TYPE | RUNTIME | FEATURES
-# | MAVEN  | APP  | JAVA    | aws-lambda,aws-cdk,amazon-api-gateway
-############
-
-./gradlew micronaut-cli:run --args="create-app --jdk 17 -b maven -f \"aws-lambda,aws-cdk,amazon-api-gateway\" temp" || EXIT_STATUS=$?
-
-if [ $EXIT_STATUS -ne 0 ]; then
-  exit $EXIT_STATUS
-fi
-
-cd starter-cli/temp
-
-./test-lambda.sh  || EXIT_STATUS=$? 
-
-if [ $EXIT_STATUS -ne 0 ]; then
-  echo "❌ FAILED | MAVEN  | APP  | JAVA    | aws-lambda,aws-cdk,amazon-api-gateway"
   exit $EXIT_STATUS
 fi
 
@@ -177,31 +101,6 @@ cd starter-cli/temp
 
 if [ $EXIT_STATUS -ne 0 ]; then
   echo "❌ FAILED | GRADLE | FUNCTION | JAVA    | aws-lambda,aws-cdk,amazon-api-gateway"
-  exit $EXIT_STATUS
-fi
-
-cd infra
-cdk destroy -f
-cd ../../..
-rm -rf starter-cli/temp
-
-############
-# | BUILD  | TYPE     | RUNTIME | FEATURES
-# | MAVEN | FUNCTION  | JAVA    | aws-lambda,aws-cdk,amazon-api-gateway
-############
-
-./gradlew micronaut-cli:run --args="create-function-app --jdk 17 -b maven -f \"aws-lambda,aws-cdk,amazon-api-gateway\" temp" || EXIT_STATUS=$?
-
-if [ $EXIT_STATUS -ne 0 ]; then
-  exit $EXIT_STATUS
-fi
-
-cd starter-cli/temp
-
-./test-lambda.sh  || EXIT_STATUS=$? 
-
-if [ $EXIT_STATUS -ne 0 ]; then
-  echo "❌ FAILED | MAVEN | FUNCTION  | JAVA    | aws-lambda,aws-cdk,amazon-api-gateway"
   exit $EXIT_STATUS
 fi
 

